v0/destinations/salesforce/transform.js

```diff
@@ -2,171 +2,184 @@ const get = require("get-value");
-    }&client_secret=${destination.Config.consumerSecret}&grant_type=password`,
-    {}
-  );
-
-  return [`Bearer ${response.data.access_token}`, response.data.instance_url];
-}
-
-// not supported from the dashboard
-function getParamsFromConfig(message, destination) {
-  const params = {};
-
-  const obj = {};
-  // customMapping: [{from:<>, to: <>}] , structure of custom mapping
-  if (destination.Config.customMappings) {
-    destination.Config.customMappings.forEach(mapping => {
-      obj[mapping.from] = mapping.to;
-    });
-  }
-  const keys = Object.keys(obj);
-  keys.forEach(key => {
-    params[obj[key]] = get(message.properties, key);
-  });
-  return params;
+  const authUrl = `${SF_TOKEN_REQUEST_URL}?username=${
+    destination.Config.userName
+  }&password=${encodeURIComponent(
+    destination.Config.password
+  )}${encodeURIComponent(destination.Config.initialAccessToken)}&client_id=${
+    destination.Config.consumerKey
+  }&client_secret=${destination.Config.consumerSecret}&grant_type=password`;
+  const response = await axios.post(authUrl, {});
+
+  return {
+    token: `Bearer ${response.data.access_token}`,
+    instanceUrl: response.data.instance_url
+  };
 }
 
 // Basic response builder
 // We pass the parameterMap with any processing-specific key-value prepopulated
 // We also pass the incoming payload, the hit type to be generated and
 // the field mapping and credentials JSONs
-async function responseBuilderSimple(
-  message,
-  mappingJson,
-  ignoreMapJson,
-  destination,
-  targetEndpoint,
-  authorizationData
-) {
-  // First name and last name need to be extracted from the name field
-  // and inserted into the message
-  let firstName = "";
-  let lastName = "";
-  const traits = getFieldValueFromMessage(message, "traits");
-  if (traits.name) {
-    // Split by space and then take first and last elements
-    const nameComponents = traits.name.split(" ");
-    firstName = nameComponents[0]; // first element
-    lastName = nameComponents[nameComponents.length - 1]; // last element
-    // Insert first and last names separately into message
-    traits.firstName = firstName;
-    traits.lastName = lastName;
+function responseBuilderSimple(message, salesforceMap, authorizationData) {
+  const { salesforceType, salesforceId } = salesforceMap;
+
+  // if id is valid, do update else create the object
+  // POST for create, PATCH for update
+  let targetEndpoint = `${authorizationData.instanceUrl}/services/data/v${SF_API_VERSION}/sobjects/${salesforceType}`;
+  if (salesforceId) {
+    targetEndpoint += `/${salesforceId}?_HttpMethod=PATCH`;
   }
 
-  const rawPayload = constructPayload(message, mappingJson);
+  // First name and last name need to be extracted from the name field
+  // get traits from the message
+  let traits = getFieldValueFromMessage(message, "traits");
+  if (traits) {
+    let rawPayload = traits;
+    // map using the config only if the type is Lead
+    if (salesforceType === "Lead") {
```

Comment:
can we add a link to SF doc for this feature

Reply:
Here is the documentation: https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_lead.htm?search_text=Lead

But it's because of backward compatibility